src/main/java/ca/wolfram/beta/symath/VMap.java

```diff
@@ -0,0 +1,24 @@
+package ca.wolfram.beta.symath;
+import java.util.HashMap;
+
+/**
+ * Our own hashmap implementation to warn users when overwriting values
+ */
+public class VMap extends HashMap<String, Double> {
+    @Override
+    public Double put(String key, Double value) {
+        if (containsKey(key))
```

Comment:
maybe only throw warnings if the value is different. if the value is the same i don't think we need to throw a warning. Otherwise everytime i create a map constant pi it will throw a warning saying i am replacing pi with pi. 

I wonder if vmap is the wrong thing to use for MathConstantNode. Instead it should have a map where the values just never change. I think we can put that as a TODO